# HG changeset patch
# User Daniel J Sebald <daniel.sebald@ieee.org>
# Date 1521185370 18000
#      Fri Mar 16 02:29:30 2018 -0500
# Node ID 75fef3484d12c51c46d0f20dc46ae963da3025ef
# Parent  a656665b671dee200641a91ee9fe106c58c8f728
Prevent hidden predecessor windows appearing when successor is floated (bug #53357)

* octave-dock-widget.cc (octave_dock_widget::set_focus_predecessor): Only
  refocus the predecessor if it is visible.

diff --git a/libgui/src/octave-dock-widget.cc b/libgui/src/octave-dock-widget.cc
--- a/libgui/src/octave-dock-widget.cc
+++ b/libgui/src/octave-dock-widget.cc
@@ -682,17 +682,18 @@ namespace octave
                                         css_background));
       }
   }
 
   // set focus to previously active widget in tabbed widget stack
   void
   octave_dock_widget::set_focus_predecessor (void)
   {
-    if (m_predecessor_widget)    // only != 0 if widget was tabbed
+    // only != 0 if widget was tabbed
+    if (m_predecessor_widget && m_predecessor_widget->isVisible ())
       m_predecessor_widget->focus ();
 
     m_predecessor_widget = nullptr;
     // FIXME: Until cset bda0c5b38bda, the wrong keys "Dockwidget/..." were used
     // here.  This had no effect in Qt4, but does in Qt5.  In the following, the
     // four incorrect keys are updated if still present in the settings files.
     // The keys are also used in the settings dialog, but
     // octave_dock_widget::handle_settings is already called at program start.
